src/Avalonia.Controls/Window.cs

```diff
@@ -449,13 +449,13 @@ public override void Show()
             EnsureInitialized();
             IsVisible = true;
             LayoutManager.ExecuteInitialLayoutPass(this);
+            SetWindowStartupLocation(Owner?.PlatformImpl);
 
             using (BeginAutoSizing())
             {
                 PlatformImpl?.Show();
```

Comment:
I believe that's where we compute the size for the window. So using SetWindowStartupLocation before the size is computed will give invalid position with auto-sized window

Reply:
It looks like `LayoutManager.ExecuteInitialLayoutPass(this);` includes auto-sizing logic. I've included a unit test for this. I tested this out with the repro, and although `ClientSize` is set again by `ShowDialog` (in the `BeginAutoSizing` section), its value hasn't changed since the first time it was set.